test/SetHostAuto.cpp

```diff
@@ -115,6 +115,26 @@ TEST(SetHostAuto, NullValueApplied) {
     uriFreeUriMembersA(&uri);
 }
 
+TEST(SetHostAuto, NullValueAppliedNoPath) {
+    UriUriA uri = parseWellFormedUri("https://example.com");
+
+    EXPECT_EQ(uriSetHostAutoA(&uri, NULL, NULL), URI_SUCCESS);
+
+    assertUriEqual(&uri, "https:");
+
+    uriFreeUriMembersA(&uri);
+}
+
+TEST(SetHostAuto, NullValueAppliedNoPath2) {
+    UriUriA uri = parseWellFormedUri("https://example.com?foo=bar");
+
+    EXPECT_EQ(uriSetHostAutoA(&uri, NULL, NULL), URI_SUCCESS);
+
+    assertUriEqual(&uri, "https:?foo=bar");
+
+    uriFreeUriMembersA(&uri);
+}
```

Comment:
@TimWolla is this test adding any value that `NullValueAppliedNoPath` right above does not already add? I'm consdering to ask you top drop that second test but maybe there is an idea that I am missing?

Reply:
> is this test adding any value that `NullValueAppliedNoPath` right above does not already add?

I added it to verify that “anything after the path” will not change the behavior. I can drop either of the tests if you'd like (or keep them both if you agree with the intention).